### ISSUE TITLE
R Markdown - knitting to bookdown::word_document2 fails because of officer::ftext used in inline code

### ISSUE THREAD

**Issue by AnxiousLobster** [davidgohel/officer#402]
Copy of stackoverflow post:
https://stackoverflow.com/questions/70652771/r-markdown-knitting-to-bookdownword-document2-fails-because-of-officerftex

Minimum Reproducible example of my R notebook (.Rmd) file which causes the error while knitting to word document:

```
---
title: "R Notebook"
output: bookdown::word_document2
---

Using inline R code chunks with `officer::ftext` causes an error as such


`r officer::ftext('Example text', officer::fp_text(color = 'red'))`
```

The error when knitting is as follows:

```
Error in vapply(x, format_sci_one, character(1L), ..., USE.NAMES = FALSE) : values must be length 1, but FUN(X[[2]]) results in length 8 Calls: <Anonymous> ... paste -> hook -> .inline.hook -> formact_sci -> vapply
```

This error started occuring only recently, previously with this code I was able to output a word document where the inline code chunk would produce text 'Example text' with red font color.

My sessionInfo() output is:

```
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=German_Switzerland.1252  LC_CTYPE=German_Switzerland.1252    LC_MONETARY=German_Switzerland.1252 LC_NUMERIC=C                       
[5] LC_TIME=German_Switzerland.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:  [1] ftExtra_0.0.3     flextable_0.5.10  officer_0.3.11    bookdown_0.24     knitr_1.37        rmarkdown_2.11    mgsub_1.7.3       lubridate_1.8.0   readxl_1.3.1      [10] data.table_1.14.2 scales_1.1.1      ggplot2_3.3.5     tidyr_1.1.4       dplyr_1.0.7       ROracle_1.3-1.1   DBI_1.1.2

loaded via a namespace (and not attached):  [1] zip_2.0.3         Rcpp_1.0.7        pillar_1.6.4      compiler_3.5.1    cellranger_1.1.0  base64enc_0.1-3   tools_3.5.1       uuid_1.0-3        digest_0.6.29     [10] evaluate_0.14     lifecycle_1.0.1   tibble_3.1.6      gtable_0.3.0      pkgconfig_2.0.3   rlang_0.4.12      yaml_2.2.1        xfun_0.29         fastmap_1.1.0     [19] xml2_1.3.3        withr_2.4.3       systemfonts_1.0.3 gdtools_0.2.3     generics_0.1.1    vctrs_0.3.8       grid_3.5.1        tidyselect_1.1.1  glue_1.6.0
[28] R6_2.5.1          fansi_0.5.0       purrr_0.3.4       magrittr_2.0.1    ellipsis_0.3.2    htmltools_0.5.2   colorspace_2.0-2  utf8_1.2.2        munsell_0.5.0     [37] crayon_1.4.2
```

I'm using Pandoc 2.1.  
RStudio version is 1.1.453.  
Please note that I'm using this R setup at work, so I'm not able to update to newer version of R or RStudio IDE.

One stackoverflow user commented that he is using R 4.1.2 and RStudio Desktop on Ubuntu so the bug does not seem to be specific to R, RStudio or OS.

**Comment by davidgohel** [comment]
I can't find how it was working before? This is weird as this code can work as you expect (in my mind at least). Are you sure you used the exact same code before?

**Comment by AnxiousLobster** [comment]
Hi David,

I'm sure the syntax is the same, I was knitting the exact same .Rmd file which used officer::ftext before 01.01.2022 and it was producing the expected output in terms of text formatting in word. Only when I tried knitting last week did it started producing the error. I also tried wrapping it in office::fpar as such:

`officer::fpar(officer::ftext('Example text', officer::fp_text(color = 'red')))`

but the same error appears.

**Comment by davidgohel** [comment]
If your environment did not change (I can see you also are still using old version of packages officer and flextable), how could officer be the source of your issue? 

knitr print methods have always been in package officedown, so really I can't see how it was working with the code you shared. I think you deleted `library(officedown)` from your Rmd.

Could you add this in an R chunk code and check it does work?

```
library(officedown)
```

**Comment by AnxiousLobster** [comment]
Thanks a lot for your advice, it worked!
The reason I have the old versions of `officer` and `flextable` is due to the version of R on my corporate machine I was able to install versions of dependency packages that only allowed to install the version of `officer` and `flextable` that you see in my post.
I installed `officedown` version 0.1.0 and after loading it the .Rmd started knitting properly again.

I tried first with `officedown` version 0.2.0, but with that version when I tried using `officer::ftext` with non-alphanumeric text input (eg. '%' or '[') instead of formatted text it returned an xml tree in the word doc.

But anyway it worked, thanks again for your help!
